test/services/SubmissionServiceSpec.scala

```diff
@@ -41,156 +43,286 @@ class SubmissionServiceSpec
+
+        result mustBe JourneyCompleted(journeyConfigWithoutBV.continueUrl)
+
+        verifyStoreRegistrationResponse(testJourneyId, RegistrationNotCalled)
+
+      }
+    }
+
+    "the user has no Nino and disable(EnableNoNinoJourney)" should {
+      s"not register and not create a trn"  in  {
+
+        val matchingResult = Random.nextBoolean()
+
+        disable(EnableNoNinoJourney)
+        val journeyConfigWithoutBV = testSoleTraderJourneyConfig.copy(businessVerificationCheck = false)
+
+        mockRetrieveIndividualDetails(testJourneyId)(Future.successful(Some(testIndividualDetailsNoNinoNoSautr)))
+        mockMatchSoleTraderDetails(testJourneyId, testIndividualDetailsNoNinoNoSautr, journeyConfigWithoutBV)(Future.successful(Right(matchingResult)))
+        mockStoreRegistrationResponse(testJourneyId, RegistrationNotCalled)(Future.successful(SuccessfullyStored))
+
+        val result = await(TestService.submit(testJourneyId, journeyConfigWithoutBV))
+
+        result mustBe JourneyCompleted(journeyConfigWithoutBV.continueUrl)
+
+        verifyStoreRegistrationResponse(testJourneyId, RegistrationNotCalled)
+
+      }
+    }
+
+    "there is a matching error" should {
+      s"return a SoleTraderDetailsMismatch"  in  {
+
+        val journeyConfigWithoutBV = testSoleTraderJourneyConfig.copy(businessVerificationCheck = false)
+
+        mockRetrieveIndividualDetails(testJourneyId)(Future.successful(Some(testIndividualDetails)))
+        mockMatchSoleTraderDetails(testJourneyId, testIndividualDetails, journeyConfigWithoutBV)(Future.successful(Left(DetailsMismatch)))
+        mockStoreIdentifiersMatch(testJourneyId, identifiersMatch = true)(Future.successful(SuccessfullyStored))
+
+        mockStoreRegistrationResponse(testJourneyId, RegistrationNotCalled)(Future.successful(SuccessfullyStored))
+
+        val result = await(TestService.submit(testJourneyId, journeyConfigWithoutBV))
+
+        result mustBe SoleTraderDetailsMismatch(DetailsMismatch)
+
+        verifyStoreRegistrationResponse(testJourneyId, RegistrationNotCalled)
+      }
+    }
+
+    "there is no matching" should {
+      s"return a JourneyCompleted"  in  {
+
+        val journeyConfigWithoutBV = testSoleTraderJourneyConfig.copy(businessVerificationCheck = false)
+
+        mockRetrieveIndividualDetails(testJourneyId)(Future.successful(Some(testIndividualDetails)))
+        mockMatchSoleTraderDetails(testJourneyId, testIndividualDetails, journeyConfigWithoutBV)(Future.successful(Right(false)))
+        mockStoreIdentifiersMatch(testJourneyId, identifiersMatch = true)(Future.successful(SuccessfullyStored))
+
+        mockStoreRegistrationResponse(testJourneyId, RegistrationNotCalled)(Future.successful(SuccessfullyStored))
+
+        val result = await(TestService.submit(testJourneyId, journeyConfigWithoutBV))
+
+        result mustBe JourneyCompleted(journeyConfigWithoutBV.continueUrl)
+
+        verifyStoreRegistrationResponse(testJourneyId, RegistrationNotCalled)
+      }
```

Comment:
I don't think this is a valid case. But not 100% sure what it's doing from the description. The method being mocked by mockMatchSoleTraderDetails() will never return a Right(false)

Reply:
I think line 166 is useless :( ... that mocked call is never called .... a mistake :(. I removed it

Reply:
I think the meaning is ... if there are no matching ..... (I dont care about nino no nino ... sautr no sautr ) I will return journey completed